app/uk/gov/hmrc/exports/services/NotificationService.scala

```diff
@@ -87,4 +92,84 @@ class NotificationService @Inject()(
+        val mrn = (singleResponseXml \ "Declaration" \ "ID").text
+        val formatter304 = DateTimeFormatter.ofPattern("yyyyMMddHHmmssX")
+        val dateTimeIssued =
+          LocalDateTime.parse((singleResponseXml \ "IssueDateTime" \ "DateTimeString").text, formatter304)
+        val functionCode = (singleResponseXml \ "FunctionCode").text
+
+        val nameCode =
+          if ((singleResponseXml \ "Status").nonEmpty)
+            Some((singleResponseXml \ "Status" \ "NameCode").text)
+          else None
+
+        val errors = buildErrorsFromRequest(singleResponseXml)
+
+        Notification(
+          actionId = notificationApiRequestHeaders.conversationId.value,
+          mrn = mrn,
+          dateTimeIssued = dateTimeIssued,
+          status = SubmissionStatus.retrieve(functionCode, nameCode),
+          errors = errors,
+          payload = notificationXml.toString
+        )
+      }
+
+    } catch {
+      case exc: Throwable =>
+        logger.error(s"There is a problem during parsing notification with exception: ${exc.getMessage}")
+        Seq.empty
+    }
+
+  def buildErrorsFromRequest(singleResponseXml: Node): Seq[NotificationError] =
+    if ((singleResponseXml \ "Error").nonEmpty) {
+      val errorsXml = singleResponseXml \ "Error"
+      errorsXml.map { singleErrorXml =>
+        val validationCode = (singleErrorXml \ "ValidationCode").text
+        val pointer =
+          if ((singleErrorXml \ "Pointer").nonEmpty) buildErrorPointers(singleErrorXml) else Pointer(Seq.empty)
+        NotificationError(validationCode, pointer)
+      }
+    } else Seq.empty
+
+  def buildErrorPointers(singleErrorXml: Node): Pointer = {
+    val pointersXml = singleErrorXml \ "Pointer"
+
+    val pointerSections = pointersXml.flatMap { singlePointerXml =>
+      /**
+        * Document Section Code contains section code e.g. 42A, 67A.
+        * One section is one element in the declaration tree e.g. Declaration, GoodsShipment etc. - non optional
+        */
+      val documentSectionCode: Option[PointerSection] =
+        Some(PointerSection((singlePointerXml \ "DocumentSectionCode").text, PointerSectionType.FIELD))
+
+      /**
+        * Sequence Numeric define what item is related with error, this is for future implementation - optional
+        */
+      val sequenceNumeric: Option[PointerSection] =
+        if ((singlePointerXml \ "SequenceNumeric").nonEmpty)
+          Some(PointerSection((singlePointerXml \ "SequenceNumeric").text, PointerSectionType.SEQUENCE))
+        else None
+
+      /**
+        * Probably the last element in pointers, is it Important for us? - optional
+        */
+      val tagId: Option[PointerSection] =
+        if ((singlePointerXml \ "TagID").nonEmpty)
+          Some(PointerSection((singlePointerXml \ "TagID").text, PointerSectionType.FIELD))
+        else None
+
+      List(documentSectionCode, sequenceNumeric, tagId).flatten
+    }.toList
+
+    Pointer(pointerSections)
```

Comment:
Basically @PatrykRudnicki this is where you would call the `WCOPointerMappingService` to convert that `Pointer` to an Exports one

Reply:
Yes, I changed it